viewing_party/party.py

```diff
@@ -1,23 +1,109 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie = {}
+    if title and genre and rating: 
+        movie["title"] = title
+        movie["genre"] = genre
+        movie["rating"] = rating
+        return movie
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if title in movie.values():
+            user_data["watchlist"].remove(movie)
+            user_data = add_to_watched(user_data, movie)
+    return user_data

```

Comment:
We are removing a value from a list we are actively iterating over, so after the first removal, we skip a value because the way iterators work (recall our discussion on the "danger zone" from the roundtable on "Iterating over Data"). Given the data for our tests, we know this isn't a concern for this application, but what if we were working with data that had duplicate titles?

Another consideration: From a performance/time standpoint, consider if we were dealing with a list of 1,000 movies. If the item we wanted to swap happens to be the first item, we would continue looping over 999 titles before the loop exits. Exiting early also helps us avoid extra time and calculations that don't add to our final result.